File: OOP/bank.py
#withdrawal,deposit,balance,CreateAccount

from datetime import datetime
class Bank:
    def create_account(self,acno,person_name,min_balance,bank_name):
        self.acno=acno
        self.p_name=person_name
        self.min_bal=min_balance
        self.bank_name=bank_name
    def deposit(self,amount):
        self.min_bal+=amount
        print("Your A\c ",self.acno,"has been credited with amt",amount,"on",datetime.now(),"avl balance ",self.min_bal)
    def withdrawal(self,amount):
        if(amount<self.min_bal):
            self.min_bal-=amount
            print("Your A\c ", self.acno, "has been debited with amt", amount, "on", datetime.now(), "avl balance ",
                  self.min_bal)
        else:
            print("Transaction error due to insufficient balance")

    def balance(self):
        print("A\c balance is ",self.min_bal)


obj1=Bank()
obj1.create_account(5262635,"Ajay",3954,"federal")
obj1.withdrawal(200)

File: JAVASCRIPT/oop/workshop_OOP/bank.js
class Bank{
    bank_name="federal Bank"
    createAccount(accno,name,minbalnce){
        this.accno=accno
        this.name=name
        this.minbalnce=minbalnce
    }
    withdraw(amount){
        if(amount>this.minbalnce){
            console.log("Insufficient balance on our account");
        }
        else{
            this.minbalnce-=amount
            console.log(Bank.bank_name` your account ${this.accno} has been debited with ammount of ${amount} and current balance is ${this.minbalnce}`);
        }
    }
    deposit(depo){
        this.minbalnce+=depo
        console.log(`your account ${this.accno} has been credited with ammount of ${depo} and current balance is ${this.minbalnce}`);
    }
    balanceEnqry(){
        console.log(`your account ${this.accno} and current balance is ${this.minbalnce}`);
    }
}

// obj1=new Bank
// obj1.createAccount(101010,"Akshay",2000,"fedaral Bank")
// obj1.deposit(12599)
obj1=new Bank
obj1.createAccount(101010,"Akshay",2000)
obj1.deposit(599)